src/features/pastelDB/pastelDBThread.tsx

```diff
@@ -0,0 +1,227 @@
+  const result = await getRpcData(props.rpcConfig)
+  insertTxoutsetinfo(props.pastelDB, result)
+}
+
+const fetchChaintips = async (props: fetchFuncConfig) => {
+  const chaintips = await getChaintips(props.rpcConfig)
+  for (let i = 0; i < chaintips.length; i++) {
+    insertChaintips(props.pastelDB, chaintips[i])
+  }
+}
+
+const fetchBlocksubsidy = async (props: fetchFuncConfig) => {
+  const result = await getBlockSubSidy(props.rpcConfig)
+  insertBlocksubsidy(props.pastelDB, result)
+}
+
+const fetchWalletinfo = async (props: fetchFuncConfig) => {
+  const result = await getWalletInfo(props.rpcConfig)
+  insertWalletinfo(props.pastelDB, result)
+}
+
+const fetchListTransactions = async (props: fetchFuncConfig) => {
+  const transactions = await listTransactions(props.rpcConfig)
+  for (let i = 0; i < transactions.length; i++) {
+    insertListTransactions(props.pastelDB, transactions[i])
+  }
+}
+
+const fetchListunspent = async (props: fetchFuncConfig) => {
+  const unspentlist = await listUnspent(props.rpcConfig)
+  for (let i = 0; i < unspentlist.length; i++) {
+    insertListunspent(props.pastelDB, unspentlist[i])
+  }
+}
+
+const fetchTotalbalance = async (props: fetchFuncConfig) => {
+  const result = await getTotalBalance(props.rpcConfig)
+  insertTotalbalance(props.pastelDB, result)
+}
+
+const fetchListaddresses = async (props: fetchFuncConfig) => {
+  const addresslist = await listAddresses(props.rpcConfig)
+  for (let i = 0; i < addresslist.length; i++) {
+    insertListaddresses(props.pastelDB, addresslist[i])
+  }
+}
+
+export const PastelDBThread = (
+  props: PastelDBThreadProps,
+): React.ReactElement => {
+  const { rpcConfig } = props
+  const [isStarted, setStarted] = useState(false)
+
+  useEffect(() => {
+    const pastelDB = remote.getGlobal('pastelDB')
+    const fetchStatisticDataFromRPC = async () => {
+      if (pastelDB && rpcConfig && rpcConfig.username !== '') {
+        // fetch whole data from RPC and save to pastel DB.
+        const pastelConfig: fetchFuncConfig = {
+          pastelDB,
+          rpcConfig,
+        }
+        setStarted(true)
+        await Promise.all([
+          fetchStatisticinfo(pastelConfig),
+          fetchNetworkinfo(pastelConfig),
+          fetchNettotals(pastelConfig),
+          fetchMempoolinfo(pastelConfig),
+          fetchRawMempoolinfo(pastelConfig),
+          fetchMininginfo(pastelConfig),
+          fetchBlock(pastelConfig),
+          fetchRawtransaction(pastelConfig),
+          fetchTransaction(pastelConfig),
+          fetchTxoutsetinfo(pastelConfig),
+          fetchChaintips(pastelConfig),
+          fetchBlocksubsidy(pastelConfig),
+          fetchWalletinfo(pastelConfig),
+          fetchListTransactions(pastelConfig),
+          fetchListunspent(pastelConfig),
+          fetchTotalbalance(pastelConfig),
+          fetchListaddresses(pastelConfig),
+        ])
+
+        setTimeout(() => fetchStatisticDataFromRPC(), 10000)
```

Comment:
```ts
        // in the beggning of the function
        let timer: number = 0;

        // here
        timer = setTimeout(() => fetchStatisticDataFromRPC(), 10000)
```
But also, this is a bit strange to me. Why is this on the timeout? Can we have some event controlling it rather an arbitrary set amount of time?

Reply:
Now we are fetching the data from RPC periodically for statistics data.
Do you mean that should fetch the data by some event? Actually, we don't have any event for this.